clients/src/main/java/org/apache/kafka/common/TopicPartition.java

```diff
@@ -28,12 +28,19 @@ public final class TopicPartition implements Serializable {
     private int hash = 0;
     private final int partition;
     private final String topic;
+    private int designatedLeader = -1;
```

Comment:
I don't think this change should be in as fundamental a class as TopicPartition. This is used all over the place in Kafka and a change here will impact efficiency (and readability)

Instead, we should create a subclass of this, that AdminClient can use. Then we have to test if the TopicPartition we're passed is one of those (yes, it's messy, but it preserves compatibility).